src/sql/workbench/api/common/extHostModelViewDialog.ts

```diff
@@ -600,25 +619,27 @@ class WizardImpl implements azdata.window.Wizard {
 	}
 }
 
-class ModelViewDashboardImpl implements azdata.window.ModelViewDashboard {
+class ModelViewDashboardImpl extends Disposable implements azdata.window.ModelViewDashboard {
 	private _tabbedPanel: azdata.TabbedPanelComponent;
 	private _view: azdata.ModelView;
 
 	constructor(
 		private _editor: ModelViewEditorImpl,
 		private _options?: azdata.ModelViewDashboardOptions
 	) {
+		super();
+		this._register(this._editor);
 	}
 
-	updateTabs(tabs: (azdata.DashboardTab | azdata.DashboardTabGroup)[]): void {
+	updateTabs(tabs: (DashboardTab | DashboardTabGroup)[]): void {
 		if (this._tabbedPanel === undefined || this._view === undefined) {
 			throw new Error(nls.localize('dashboardNotInitialized', "Tabs are not initialized"));
 		}
 
 		this._tabbedPanel.updateTabs(this.createTabs(tabs, this._view));
 	}
 
-	registerTabs(handler: (view: azdata.ModelView) => Thenable<(azdata.DashboardTab | azdata.DashboardTabGroup)[]>): void {
+	registerTabs(handler: (view: azdata.ModelView) => Thenable<(DisposableDashboardTab | DisposableDashboardTabGroup)[]>): void {
```

Comment:
The return type for this should also not change since it's the user defining the handler - and they won't be expected to know to return the Disposable tab objects from that (and we need to know internally that what we get from them isn't actually disposable)